x-pack/plugins/ingest_manager/public/plugin.ts

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import {
+  AppMountParameters,
+  CoreSetup,
+  CoreStart,
+  Plugin,
+  PluginInitializerContext,
+} from 'kibana/public';
+import { i18n } from '@kbn/i18n';
+import { DEFAULT_APP_CATEGORIES } from '../../../../src/core/utils';
+import { DataPublicPluginSetup } from '../../../../src/plugins/data/public';
+import { LicensingPluginSetup } from '../../licensing/public';
+import { PLUGIN_ID } from '../common/constants';
+import { IngestManagerConfigType } from '../common/types';
+import { BASE_PATH } from './applications/ingest_manager/constants';
+
+export { IngestManagerConfigType } from '../common/types';
+
+export type IngestManagerSetup = void;
+export type IngestManagerStart = void;
+
+export interface IngestManagerSetupDeps {
+  licensing: LicensingPluginSetup;
+  data: DataPublicPluginSetup;
+}
+
+export class IngestManagerPlugin implements Plugin {
+  private config: IngestManagerConfigType;
+
+  constructor(private readonly initializerContext: PluginInitializerContext) {
+    this.config = this.initializerContext.config.get<IngestManagerConfigType>();
+  }
+
+  public setup(core: CoreSetup, deps: IngestManagerSetupDeps) {
+    const config = this.config;
+
+    // Register main Ingest Manager app
+    core.application.register({
+      id: PLUGIN_ID,
+      category: DEFAULT_APP_CATEGORIES.management,
+      title: i18n.translate('xpack.ingestManager.appTitle', { defaultMessage: 'Ingest Manager' }),
+      appRoute: BASE_PATH,
```

Comment:
Your `appRoute` here is the default. I would suggest just not specifying one.

Reply:
I removed that in https://github.com/elastic/kibana/pull/56567/commits/a411d26591b419619c65ce0635f6ab8121ef78c3

I think I found a bug but let me know what you think. When I registered the app using
<img width="780" alt="Screen Shot 2020-02-11 at 12 41 38 PM" src="https://user-images.githubusercontent.com/57655/74285122-2c2afc80-4cf3-11ea-99e7-4b297bbfba7e.png">

It correctly added the nav item and set the route to `/app/FROM_ROUTE` based on the value I gave to `appRoute`
<img width="384" alt="Screen Shot 2020-02-11 at 12 42 19 PM" src="https://user-images.githubusercontent.com/57655/74285163-42d15380-4cf3-11ea-84b1-fe3939a51d52.png">

However, that route didn't work <img width="634" alt="Screen Shot 2020-02-11 at 12 45 59 PM" src="https://user-images.githubusercontent.com/57655/74285207-64323f80-4cf3-11ea-9344-ad69c4c93633.png">

`/app/FROM_ID` did though. After looking at this section https://github.com/elastic/kibana/blob/f890776e6dbc353132f2d392447b77951498dad3/src/core/public/application/ui/app_router.tsx#L51-L53 there seems to be conflict between creating routes to `/app/:customRoute` but later requiring anything under `/app` to use the app's id.

This isn't a huge issue for us. Let me know if there's something I'm missing to get this to go to `/app/ingest` for example

Reply:
So I do think there may be a bug in the custom `appRoute` code, but if you simply don't specify an `appRoute` and your application's ID is `ingest` then Core should mount your app at `/app/ingest` automatically.